prow/cmd/sidecar/main.go

```diff
@@ -18,28 +18,34 @@ package main
 
 import (
 	"context"
+	"io"
+	"os"
 
 	"github.com/sirupsen/logrus"
 
-	"k8s.io/test-infra/prow/logrusutil"
 	"k8s.io/test-infra/prow/pod-utils/options"
 	"k8s.io/test-infra/prow/sidecar"
 )
 
 func main() {
-	logrusutil.ComponentInit()
-	logrus.SetLevel(logrus.DebugLevel)
-
+	logFile, err := sidecar.LogSetup()
+	if err != nil {
+		logrus.Errorf("Unable to set up log file")
+	}
 	o := sidecar.NewOptions()
+
 	if err := options.Load(o); err != nil {
 		logrus.Fatalf("Could not resolve options: %v", err)
 	}
 
+	defer os.Remove(logFile.Name())
```

Comment:
nit: This should come immediately after the error check for `sidecar.LogSetup()` so that we still clean up the logfile if there is an error loading the options.

Reply:
the file hasn't been passed to the Run function yet though so, there will be an issue

Reply:
What is the issue?
Either way the file pointer is passed to `o.Run()` later, I'm not suggesting changing the ordering of that. I'm suggesting moving this up to between line 34 and 35.

Reply:
read this wrongly, i thought you meant removing the defer, and just removing the file i see what you meant now